### ISSUE TITLE
Not catching end-of-file-based errors

### ISSUE THREAD

**Issue by bstoutenburgh** [maristgeek/SublimeLinter-contrib-bashate#13]
```
SublimeLinter: bashate: E040_syntax_error.sh ['/usr/local/bin/bashate'] 
SublimeLinter: bashate output:
[E] E040: Syntax error: syntax error near unexpected token `}': ''
 - /tmp/SublimeLinter3-ben/E040_syntax_error.sh : L7
1 bashate error(s) found 
```

![screenshot_2016-05-14_17-53-33](https://cloud.githubusercontent.com/assets/418321/15270874/fb89307e-19fc-11e6-98b3-ad405f621e98.png)

The regex is not catching E040 at all. Confirmed in settings that E040 is not being ignored.

**Comment by bstoutenburgh** [comment]
``` python
        if 'syntax error' in m.group('error'):
            msg = '%s: %s' % (MESSAGES['E040'].msg, m.group('error'))
            report.print_error(msg, filename=filename,
                               filelineno=int(m.group('lineno')))

        # Matching output from bash warning about here-documents not
        # ending.
        # FIXME: are there other warnings that might come out
        # with "bash -n"?  A quick scan of the source code suggests
        # no, but there might be other interesting things we could
        # catch.
        if 'warning:' in m.group('error'):
            if 'delimited by end-of-file' in m.group('error'):
                start = re.match('^.*line (?P<start>[0-9]+).*$',
                                 m.group('error'))
                report.print_error(
                    MESSAGES['E012'].msg % int(start.group('start')),
                    filename=filename,
                    filelineno=int(m.group('lineno')))
```

Because bashate itself is inconsistent with its output. Compaire this to say E004...

``` python
        if not line.endswith('\n'):
            report.print_error(MESSAGES['E004'].msg, line)
```

This will make regex really complicated. But confirmed that we're currently not catching E012 for the same reason.
